app/main/views/suppliers.py

```diff
@@ -449,35 +484,35 @@ def update_supplier_framework_details(supplier_id, framework_slug):
             if not user:
                 abort(400, "No user found with id '{}'".format(update_json['agreementDetails']['uploaderUserId']))
 
-        interest_record.agreement_details = agreement_details or None
+        framework_agreement.signed_agreement_details = agreement_details or None
 
-    uniform_now = datetime.utcnow()
-
-    if 'onFramework' in update_json:
-        interest_record.on_framework = update_json['onFramework']
     if 'agreementReturned' in update_json:
         if update_json["agreementReturned"] is False:
-            interest_record.agreement_returned_at = None
-            interest_record.agreement_details = None
+            framework_agreement.signed_agreement_returned_at = None
+            framework_agreement.signed_agreement_details = None
         else:
-            interest_record.agreement_returned_at = uniform_now
-    if update_json.get('countersigned'):
-        interest_record.countersigned_at = uniform_now
+            framework_agreement.signed_agreement_returned_at = datetime.utcnow()
+
+    try:
+        db.session.add(framework_agreement)
+        db.session.flush()
+    except IntegrityError as e:
+        db.session.rollback()
+        abort(400, e.orig)
 
     audit_event = AuditEvent(
         audit_type=AuditTypes.supplier_update,
         user=updater_json['updated_by'],
-        data={'supplierId': supplier.supplier_id, 'frameworkSlug': framework_slug, 'update': update_json},
-        db_object=supplier
+        data={
+            'supplierId': supplier.id,
+            'frameworkSlug': framework.slug,
+            'update': update_json
+        },
+        db_object=framework_agreement
```

Comment:
Having a `framework_agreement` as the object of a `supplier_update` feels a bit weird. We discussed this already though I think and decided it was the best way?